src/poetry/installation/installer.py

```diff
@@ -245,12 +245,12 @@ def _do_install(self) -> int:
             locked_repository = self._locker.locked_repository()
 
             if not self._locker.is_fresh():
-                self._io.write_error_line(
-                    "<warning>"
-                    "Warning: poetry.lock is not consistent with pyproject.toml. "
+                raise ValueError(
+                    "<error>"
+                    "Error: poetry.lock is not consistent with pyproject.toml. "
                     "You may be getting improper dependencies. "
                     "Run `poetry lock [--no-update]` to fix it."
-                    "</warning>"
+                    "</error>"
```

Comment:
I assume we don't need `<error>` and `Error:` when raising an error. Further, we can remove `"You may be getting improper dependencies."` because now you can't install anymore.